src/entity2/user.ts

```diff
@@ -0,0 +1,49 @@
+import { Entity, PrimaryGeneratedColumn, Column, BaseEntity } from 'typeorm';
+
+@Entity()
+export class User extends BaseEntity {
+    @PrimaryGeneratedColumn('uuid')
+    id: string;
+
+    @Column({ unique: true })
+    username: string;
+
+    @Column({ unique: true })
+    email: string;
+
+    @Column({ nullable: true, unique: true })
+    oidc_subject: string;
+
+    @Column({ nullable: true })
+    oidc_issuer: string;
+
+    @Column({ type: 'text', nullable: true })
+    access_token: string;
+
+    @Column({ type: 'text', nullable: true })
+    refresh_token: string;
+
+    @Column({ type: 'text', nullable: true })
+    id_token: string;
+
+    @Column({ type: 'timestamp', nullable: true })
+    token_expiry: Date;
+
+    @Column({ nullable: true })
```

Comment:
See "Falsehoods programmers believe about names".
Let's just have a single name field and ask the user to populate it with their preferred text the first time they log in.

Reply:
I'm suggesting to just ignore these attributes from OIDC because they're known to be unhelpful.